pkg/lib/dialogs.tsx

```diff
@@ -153,41 +158,28 @@ export const useDialogs = () => {
 };
 
 export const WithDialogs = ({ children } : { children: React.ReactNode }) => {
-    const is_open = useRef(false); // synchronous
-    const resolveRef = useRef<((x: unknown) => void) | null>(null);
-    const rejectRef = useRef<((x: unknown) => void) | null>(null);
     const [dialog, setDialog] = useState<React.ReactNode>(null);
+    const shown = useRef<"show" | "run" | "close">("close");
+
+    function transition(expected: string, to: typeof shown.current, arg: React.ReactNode = null) {
```

Comment:
Why isn't `expected` the same enum type as `shown`/`to`? That should move runtime failures to TS compile time check failures?

Reply:
`to` needs to have the stricter type because we assign its value to `shown.current`.  `expected` is only used for comparisons.  It could probably even have a value of `unknown`.  I'm going to switch all three places to using a single `type =`.